src/ag-mock.js

```diff
@@ -0,0 +1,97 @@
f,v,Re,Ae;I&&O.log(`Network events with unknown purposes are ${F?"always allowed":`${re?"":"not "}consentable`}`);let e={defaultConsent:"Auto",configurable:!0,essential:!1,optOutSignals:["DNT"]},n={[ce]:{name:ce,description:"",defaultConsent:!0,configurable:!1,essential:!0},[k]:{name:k,description:"",defaultConsent:!1,configurable:!1,essential:!1}},t=(f=H)==null?void 0:f.types;t&&Object.keys(t).forEach(a=>{t[a]=g(g({},e),t[a])});let o=(Re=(v=H)==null?void 0:v.useDefault)!=null?Re:!0,s=g(g({},o&&{Functional:g({name:"Functionality",description:"Personalization, autofilled forms, etc."},e),Analytics:g({name:"Analytics + Performance",description:"Help us learn how our site is used and how it performs."},e),Advertising:g({name:"Targeting / Advertising",description:"Helps us and others serve ads relevant to you."},e),SaleOfInfo:Ie(g({name:"Sale of personal information",description:""},e),{optOutSignals:["GPC","DNT"]})}),t);x={useDefault:o,types:g(g({},s),n)};let c=(a,i)=>i==="boolean"||i==="string",r=(Ae=H)==null?void 0:Ae.defaultConsent;if(typeof r<"u"){let a=c(r,typeof r)?(()=>{let i={};return Object.keys(s).forEach(S=>{i[S]=r}),i})():r;Object.keys(a).forEach(i=>{var w;let S=(w=x.types)==null?void 0:w[i];S?S.defaultConsent=a[i]:ve&&O.warn(`Default consent specified for unrecognized tracking purpose: ${JSON.stringify(i)}`)})}Ce&&(I&&O.log("Using embedded purpose map"),Bt(Kt,new h(Ce),!0)),Object.keys(x.types).forEach(a=>{let i=x.types[a],S=i.configurable&&!i.essential&&(!Y||b.has(a));Ve.add(a),U.set(a,i),S&&(qe.push(a),J.add(a)),i.configurable&&(ze.set(a,"defaultConsent"in i?We(a,i.defaultConsent):!1),Ut.set(a,"defaultConsent"in i?We(a,i.defaultConsent,!0):!1))}),Je(x);let l=a=>{var w;if(!He)return;let{purposes:i}=a,S;for(let[$,gt]of Object.entries(i))gt&&((w=U.get($))!=null&&w.configurable)&&Oe($)&&(S=!0,i[$]=!1);S&&(a.timestamp=new Date().toISOString(),a.confirmed=a.updated=!0)};for(let[a,i]of ze.entries())W[a]=i;let u=xt(),{confirmed:P,purposes:T}=u;V={purposes:W,confirmed:y,updated:R=R||!!(T&&te(Array.from(Ve),T,W)),prompted:E};let d=P?u:V;d!==V&&l(V),l(d),C=d==null?void 0:d.purposes,Ze(d)};var{airgap:p,transcend:Be}=window,it=p==null?void 0:p.macroregions,je=(Array.isArray(it)?it:ke).map(([e,n])=>[e,n.split(" ")]),ct=p==null?void 0:p.regionRegimesMap,Pe=Array.isArray(ct)?ct:Ne,pt=p==null?void 0:p.regimePurposeOptOuts,at=Array.isArray(pt)?pt:X,kn=m.regimePrecedence?m.regimePrecedence.split(";"):[],ut=p==null?void 0:p.regimePurposeScopes,Y=m.regimePurposeScopes!=="off",z=Array.isArray(ut)?ut:Q,b=me(z),Ee=b.values().next().done,H=p==null?void 0:p.purposes,Ce=p==null?void 0:p.purposeMap,lt=()=>{zt()},Yt=!1;var zt=()=>{Ke(),rt();let e={};e.getRegimePurposes=n=>me(z,n),e.getConsent=Se,e.setConsent=q,e.setConsentMetadata=Qe,e.optIn=ot,e.optOut=st,e.isOptedIn=et,e.isOptedOut=tt,e.getPurposeTypes=$e,e.getRegimes=()=>new Set(N()),e.getPrivacySignals=()=>new Set(B()),Object.defineProperty(window,"airgap",{value:e,configurable:!1,enumerable:!1}),Yt=!0};lt();
```

Comment:
can use the unminified version of this?

Reply:
if this is getting checked into git id rather keep it minified, unminified is 118kb (vs minified 15kb). i have a reference branch in main that im keeping up so people can check out unminified code (or edit and use to rebuild this stub)

Reply:
I'd agree with eli here, this should probs be unminified. Code in the test folders should not be included in whatever bundle we generate anyway, so file size shouldn't be a concern here